### PR TITLE
WDP-83

### PR DESCRIPTION
**Pull request by lukaszcwiklinski**
https://kindergarten.stxnext.pl/browse/WDP-83

### CODE REVIEW

File: src/presence_analyzer/script.py [hunk 1]
```diff
@@ -111,3 +117,22 @@ def action_stop(dry_run=False):
...
+
+def download_xml():
+    """
+    Downloads users.xml into data directory
+    :return:
+    """
+    try:
+        print "downloading with urllib"
+        urllib.urlretrieve(
+            'http://sargo.bolt.stxnext.pl/users.xml',
```
Comment (dpolcyn) [inline src/presence_analyzer/script.py]: use format

File: src/presence_analyzer/templates/mean_time_weekday.html [hunk 2]
```diff
@@ -4,29 +4,38 @@
...
         $(document).ready(function(){
-            var $loading = $('#loading');
+            var $loading = $('#loading'),
+                avatarDict = {};
 
             $.getJSON('/api/v1/users', function(result) {
                 var $dropdown = $('#user-id');
 
                 $.each(result, function(item) {
+                    avatarDict[this.user_id] = this.avatar;
```
Comment (dpolcyn) [inline src/presence_analyzer/templates/mean_time_weekday.html]: why avatarDict?

File: src/presence_analyzer/utils.py [hunk 3]
```diff
@@ -8,8 +8,10 @@
 from functools import wraps
 from datetime import datetime
 
+from StringIO import StringIO
```
Comment (dpolcyn) [inline src/presence_analyzer/utils.py]: sort imports

File: src/presence_analyzer/templates/mean_time_weekday.html (hunk 2 above)
Comment (lukaszcwiklinski) [inline src/presence_analyzer/templates/mean_time_weekday.html]: I used avatarDict to containt avatar link to assigned to specific user (user_id). It's more optimized than iterate over all items on every change.

File: src/presence_analyzer/templates/base.html [hunk 4]
```diff
@@ -27,6 +27,7 @@
                 <select id="user-id" style="display: none">
                     <option value="">--</option>
                 </select>
+                <img id="avatar" src="" alt = "" style="display: none"/>
```
Comment (piotrgrundas) [inline src/presence_analyzer/templates/base.html]: ` alt=""`

File: src/presence_analyzer/utils.py [hunk 5]
```diff
@@ -71,6 +72,52 @@ def get_data():
     return data
 
 
+def get_data_xml():
```
Comment (piotrgrundas) [inline src/presence_analyzer/utils.py]: `get_xml_data`

File: src/presence_analyzer/utils.py [hunk 6]
```diff
@@ -71,6 +72,52 @@ def get_data():
...
+                name = user.find('name')
+                avatar = user.find('avatar')
+            except (ValueError, TypeError):
+                log.debug('Problem with line %d: ', i, exc_info=True)
+
+            data.append(
+                {
+                    'user_id': int(user_id),
+                    'name': name.text,
+                    'avatar': https.text + '://' + url.text + avatar.text
```
Comment (piotrgrundas) [inline src/presence_analyzer/utils.py]: 1) declare this above for without avatar, then here just add it here
2) format...

File: src/presence_analyzer/tests.py [hunk 7]
```diff
@@ -174,12 +186,13 @@ def test_api_users(self):
         self.assertEqual(resp.status_code, httplib.OK)
         self.assertEqual(resp.content_type, 'application/json')
         data = json.loads(resp.data)
```
Comment (piotrgrundas) [inline src/presence_analyzer/tests.py]: enter and after `resp = self.client.get('/api/v1/users')`

File: src/presence_analyzer/tests.py [hunk 8]
```diff
@@ -174,12 +186,13 @@ def test_api_users(self):
         self.assertEqual(resp.status_code, httplib.OK)
         self.assertEqual(resp.content_type, 'application/json')
```
Comment (piotrgrundas) [inline src/presence_analyzer/tests.py]: enter

File: src/presence_analyzer/templates/presence_weekday.html [hunk 9]
```diff
@@ -4,34 +4,46 @@
...
     (function($) {
         $(document).ready(function(){
-            var $loading = $('#loading');
+            var $loading = $('#loading'),
+                avatarDict = {};
 
             $.getJSON('/api/v1/users', function(result) {
                 var $dropdown = $('#user-id');
 
                 $.each(result, function(item) {
```
Comment (piotrgrundas) [inline src/presence_analyzer/templates/presence_weekday.html]: item is unused since ur using `this` :)

File: src/presence_analyzer/script.py [hunk 10]
```diff
@@ -111,3 +116,20 @@ def action_stop(dry_run=False):
...
+def download_xml():
+    """
+    Downloads users.xml into data directory
+    :return:
+    """
+    try:
+        print '{}'.format('downloading with urllib')
+        urllib.urlretrieve(
+            '{}'.format('http://sargo.bolt.stxnext.pl/users.xml',
+            (XML_PATH + '{}' + '{}').format('/', 'users.xml'))
```
Comment (piotrgrundas) [inline src/presence_analyzer/script.py]: style...
`/` should be in string not passed as a param

File: src/presence_analyzer/script.py [hunk 11]
```diff
@@ -111,3 +116,20 @@ def action_stop(dry_run=False):
         _serve('stop', dry_run=dry_run)
 
     werkzeug.script.run()
+
+
+def download_xml():
+    """
+    Downloads users.xml into data directory
+    :return:
```
Comment (piotrgrundas) [inline src/presence_analyzer/script.py]: ?

File: src/presence_analyzer/script.py [hunk 12]
```diff
@@ -111,3 +116,19 @@ def action_stop(dry_run=False):
...
+
+def download_xml():
+    """
+    Downloads users.xml into data directory
+    """
+    try:
+        print 'downloading with urllib'
+        urllib.urlretrieve(
+            'http://sargo.bolt.stxnext.pl/users.xml',
+            ('{}/{}').format(XML_PATH, 'users.xml')
```
Comment (piotrgrundas) [inline src/presence_analyzer/script.py]: `'xx'.format`

File: src/presence_analyzer/utils.py [hunk 13]
```diff
@@ -71,6 +72,49 @@ def get_data():
...
+        https = server.find('protocol')
+        url = server.find('host')
+        users = xml_data.find('users')
+
+        try:
+            data = [
+                {
+                    'user_id': int(user.get('id')),
+                    'name': user.find('name').text,
+                    'avatar': '{}://{}{}'.format(
```
Comment (piotrgrundas) [inline src/presence_analyzer/utils.py]: U were told before, move up already created url and here just add avatar to it

**Pull request merged**